unity/Assets/Scripts/PhysicsRemoteFPSAgentController.cs

```diff
@@ -7051,80 +7116,98 @@ public bool objectIsCurrentlyVisible(SimObjPhysics sop, float maxDistance)
+            for (float h = -maxUpwardLookAngle; h <= maxDownwardLookAngle; h += horizon) {
+                horizons.Add(h);
             }
+            List<Dictionary<string, object>> interactablePoses = getInteractablePoses(
+                objectId: objectId,
+                markActionFinished: false,
+                positions: positions,
+                horizons: horizons.ToArray()
+            );
 
-            if (!physicsSceneManager.ObjectIdToSimObjPhysics.ContainsKey(action.objectId)) {
-                errorMessage = "Object ID appears to be invalid.";
+            if (interactablePoses == null) {
                 actionFinished(false);
                 return;
             }
 
-            SimObjPhysics theObject = physicsSceneManager.ObjectIdToSimObjPhysics[action.objectId];
+            // for backwards compatibility, PositionsFromWhichItemIsInteractable returns
+            // Dictionary<string, float> instead of List<Dictionary<string, object>>,
+            // where the latter is cleaner in python.
+            Dictionary<string, List<float>> d = new Dictionary<string, List<float>>();
+            string[] keys = {"x", "y", "z", "rotation", "standing", "horizon"};
+            foreach (string key in keys) {
+                d[key] = new List<float>();
+            }
+            foreach(Dictionary<string, object> pose in interactablePoses) {
+                foreach (string key in keys) {
+                    if (key == "standing") {
+                        // standing is converted from true => 1 to false => 0, for backwards compatibility
+                        d[key].Add((bool) pose[key] ? 1 : 0);
+                    } else {
+                        // all other keys have float outputs
+                        d[key].Add((float) pose[key]);
+                    }
+                }
+            }
+            actionFinished(true, d);
+        }
 
-            int numTimesVisible = NumberOfPositionsFromWhichItemIsVisibleHelper(theObject, positions);
+        // private helper for NumberOfPositionsFromWhichItemIsVisible
+        private int numVisiblePositions(string objectId, bool markActionFinished, Vector3[] positions = null, int maxPoses = int.MaxValue) {
+            List<Dictionary<string, object>> interactablePoses = getInteractablePoses(
+                objectId: objectId,
+                positions: positions,
+                maxDistance: 1e5f,          // super large number for maximum distance!
+                horizons: new float[] {0},  // don't care about every horizon here, just horizon=0
+                markActionFinished: false,
+                maxPoses: maxPoses
+            );
+
+            // object id might have been invalid, causing failure
+            if (markActionFinished) {
+                actionFinished(success: interactablePoses != null);
+            }
+            return interactablePoses == null ? 0 : interactablePoses.Count;
+        }
```

Comment:
This action copied nearly the same functionality of `getInteractablePoses`, but returned the number instead of each pose. Here, we just modulate its functionality.

Note (cc: @Lucaweihs), this did not check for any other horizons other than 0, by default. I can have it check for others, if necessary, but I don't know the intention of only using a single horizon here.

Reply:
This isn't correct I believe. The old code used whatever horizon you were currently using, not 0.

Reply:
That's true. I guess I just figured that these tasks never looked up or down, and hence the horizon was always 0.

But it has been updated to use the current horizon.